src/gui/TimeLineWidget.cpp

```diff
@@ -304,60 +307,103 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
+		// Handles moving playhead
 		else
 		{
-			m_moveXOff = s_posMarkerPixmap->width() / 2;
+			m_action = MovePositionMarker;
+			if (event->x() - m_xOffset < s_posMarkerPixmap->width())
+			{
+				m_moveXOff = event->x() - m_xOffset;
+			}
+			else { m_moveXOff = s_posMarkerPixmap->width() / 2; }
 		}
 	}
-	else if( event->button() == Qt::LeftButton  && (event->modifiers() & Qt::ShiftModifier) )
+	else if (button == Qt::RightButton) {
+		if (mods & Qt::ShiftModifier){ m_action = rightShiftAction; }
+		else if (mods == Qt::ControlModifier){ m_action = rightCtrlAction; }
+		else { contextMenuEvent(nullptr); }
+	}
+	
+	// Calculate the clicked position as a TimePos, several actions need this
+	m_moveXOff = s_posMarkerPixmap->width() / 2;
+	const auto ticksPerPixel = TimePos::ticksPerBar() / m_ppb;
+	const TimePos t = m_begin + static_cast<int>(qMax(event->x() - m_xOffset - m_moveXOff, 0) * ticksPerPixel);
+	
+	// Translate MoveLoopClosest into left or right based on distance
+	if (m_action == MoveLoopClosest)
 	{
-		m_action = SelectSongTCO;
-		m_initalXSelect = event->x();
+		const TimePos loopMid = (m_loopPos[0] + m_loopPos[1]) / 2;
+		if (t < loopMid) { m_action = MoveLoopBegin; }
+		else { m_action = MoveLoopEnd; }
 	}
-	else if( event->button() == Qt::RightButton )
+	// If we're moving loop markers or dragging loop markers, update them
+	if (m_action == MoveLoopBegin){ m_loopPos[0] = t; }
+	else if (m_action == MoveLoopEnd){ m_loopPos[1] = t; }
+	else if (m_action == DragLoop)
 	{
-		m_moveXOff = s_posMarkerPixmap->width() / 2;
-		const TimePos t = m_begin + static_cast<int>( qMax( event->x() - m_xOffset - m_moveXOff, 0 ) * TimePos::ticksPerBar() / m_ppb );
-		const TimePos loopMid = ( m_loopPos[0] + m_loopPos[1] ) / 2;
-
-		if( t < loopMid )
-		{
-			m_action = MoveLoopBegin;
-		}
-		else if( t > loopMid )
-		{
-			m_action = MoveLoopEnd;
-		}
-
-		if( m_loopPos[0] > m_loopPos[1]  )
-		{
-			qSwap( m_loopPos[0], m_loopPos[1] );
-		}
-
-		m_loopPos[( m_action == MoveLoopBegin ) ? 0 : 1] = t;
+		const bool quant = !(mods == (Qt::ControlModifier | Qt::ShiftModifier));
+		
+		m_loopPos[0] = quant ? t.quantize(1.0) : t;
+		m_loopPos[1] = t + (quant ? TimePos::ticksPerBar() : 1);
+	}
+	
+	// Ensure that the loops beginning and end are stored in the right place
+	if (m_loopPos[0] > m_loopPos[1]) { qSwap(m_loopPos[0], m_loopPos[1]); }
+	
+	// Set initial position for actions that need it
+	if (m_action == SelectSongTCO || m_action == DragLoop)
+	{
+		m_initalXSelect = event->x();
 	}
 
-	if( m_action == MoveLoopBegin || m_action == MoveLoopEnd )
+	// Notify the user if they can disable quantization
+	if (mods & Qt::ShiftModifier)
 	{
 		delete m_hint;
 		m_hint = TextFloat::displayMessage( tr( "Hint" ),
-					tr( "Press <%1> to disable magnetic loop points." ).arg(UI_CTRL_KEY),
-					embed::getIconPixmap( "hint" ), 0 );
+			tr( "Hold <%1> to disable quantization." ).arg(UI_CTRL_KEY),
+			embed::getIconPixmap( "hint" ), 0 );
```

Comment:
```suggestion
			tr("Hold <%1> to disable quantization.").arg(UI_CTRL_KEY),
			embed::getIconPixmap("hint"), 0);
```

Plus maybe having that hint box for Ctrl actions as well as suggested in the other comment